app/group.go

```diff
@@ -102,14 +102,14 @@ func (a *App) UpsertGroupSyncable(groupSyncable *model.GroupSyncable) (*model.Gr
 			return nil, err
 		}
 		if team.IsGroupConstrained() {
-			var teamGroups []*model.Group
-			teamGroups, err = a.GetAllGroupsByTeam(team.Id)
+			var teamGroups []*model.GroupWithSchemeAdmin
+			teamGroups, err = a.Srv().Store.Group().GetGroupsByTeam(channel.TeamId, model.GroupSearchOpts{})
```

Comment:
nit: we could use `:=` here, and remove the var declaration, but up to you (since its your code)

Reply:
I think this was one of the "shadow declaration" changes.